src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -60,80 +94,457 @@ W3DDisplay::W3DDisplay() :
 // 0x0073C453
 W3DDisplay::~W3DDisplay()
 {
-    // delete m_debugDisplay;
-    // for (auto &str : m_displayStrings) {
-    //    g_theDisplayStringManager->Free_Display_String(str);
-    //}
+    delete m_debugDisplay;
 
-    // delete m_2DRender;
-    // m_2DRender = nullptr;
-    // Delete_Views();
-    // s_3DScene->Release_Ref();
-    // s_3DScene = nullptr;
-    // s_2DScene->Release_Ref();
-    // s_2DScene = nullptr;
-    // s_3DInterfaceScene->Release_Ref();
-    // s_3DInterfaceScene = nullptr;
-
-    // for (auto &light : m_myLight) {
-    //    light->Release_Ref();
-    //    light = nullptr;
-    //}
+    for (int i = 0; i < 16; i++) {
+        g_theDisplayStringManager->Free_Display_String(m_displayStrings[i]);
+    }
 
-    // PredictiveLODOptimizerClass::Free();
-    //// unk string 0xA48068 StringClass::Release_Resources();
-    // W3DShaderManager::Shutdown();
-    // s_assetManager->Free_Assets();
-    // delete s_assetManager;
-    //// WW3D::Shutdown();
-    //// WWMath::Shutdown();
-    //// Browser Shutdown();
-    //// delete g_theW3DFileSystem;
-
-    // TODO: Requires WW3D, WWMath, g_theW3DFileSystem, 0xA48068, Browser
-#ifdef GAME_DLL
-    Call_Method<void, W3DDisplay>(0x0073C470, this);
-#endif
+    delete m_2DRender;
+    m_2DRender = nullptr;
+
+    Delete_Views();
+
+    s_3DScene->Release_Ref();
+    s_3DScene = nullptr;
+
+    s_2DScene->Release_Ref();
+    s_2DScene = nullptr;
+
+    s_3DInterfaceScene->Release_Ref();
+    s_3DInterfaceScene = nullptr;
+
+    for (int i = 0; i < 4; i++) {
+        Ref_Ptr_Release(m_myLight[i]);
+    }
+
+    PredictiveLODOptimizerClass::Free();
+    // Debug_Statistics::Shutdown_Statistics
+    W3DShaderManager::Shutdown();
+    s_assetManager->Free_Assets();
+    delete s_assetManager;
```

Comment:
nullptr check

Reply:
Looked at the obj compiled by MSVC from this code and its got the nullptr check in there.